src/pivot-table/hooks/use-column-width.ts

```diff
@@ -1,167 +1,245 @@
 import { memoize } from "qlik-chart-modules";
 import { useCallback, useMemo } from "react";
 import { PSEUDO_DIMENSION_INDEX } from "../../constants";
-import type { LayoutService, LeftDimensionData, Rect, VisibleDimensionInfo } from "../../types/types";
+import {
+  ColumnWidthType,
+  type ColumnWidth,
+  type ExtendedDimensionInfo,
+  type ExtendedMeasureInfo,
+} from "../../types/QIX";
+import type { LayoutService, Rect, VisibleDimensionInfo } from "../../types/types";
 import { GRID_BORDER } from "../constants";
 import { useStyleContext } from "../contexts/StyleProvider";
 import useMeasureText from "./use-measure-text";
 
 interface ColumnWidthHook {
   leftGridWidth: number;
   rightGridWidth: number;
-  totalMeasureInfoColumnWidth: number;
-  getLeftColumnWidth: (index: number) => number;
-  getDataColumnWidth: (index: number) => number;
-  getMeasureInfoWidth: (index: number) => number;
-  getTotalWidth: () => number;
+  totalWidth: number;
+  showLastRightBorder: boolean;
+  getLeftGridColumnWidth: (index: number) => number;
+  getRightGridColumnWidth: (index?: number) => number;
 }
 
 export const EXPAND_ICON_WIDTH = 30;
-const MIN_COLUMN_WIDTH = 100;
-const MAX_RATIO_OF_TOTAL_WIDTH = 0.75;
+const LEFT_GRID_MAX_WIDTH_RATIO = 0.75;
+
+export enum ColumnWidthValues {
+  PixelsMin = 30,
+  PixelsMax = 7680,
+  PixelsDefault = 200,
+  PercentageMin = 1,
+  PercentageMax = 100,
+  PercentageDefault = 20,
+  AutoMin = 80,
+}
+
+const getValidValue = (value: number | undefined, defaultValue: number) =>
+  !!value && typeof value === "number" && !Number.isNaN(value) ? value : defaultValue;
```

Comment:
```suggestion
  !!value && !Number.isNaN(value) ? value : defaultValue;
```

if value is not falsy -> `!!value` will take care of `| undefined` case and there after the value argument will be treated as number. 
So i presume you dont need to check `typeof` it? unless typescript complains about something else

Reply:
Because strings that are only number return false from isNaN, e.g. `'1'`

Reply:
you should not have numbers as strings because you are getting value as number (`value: number`) in arguments

Reply:
Yeah if the user follows the typing. I think you can set it to a string, then we want to cover that. But maybe it is overkill